### PR TITLE
Request mapping pincode

### PR DESCRIPTION
**Pull request by jitendrajaria**
pincode request service added.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -41,6 +41,7 @@
             <artifactId>commons-lang3</artifactId>
             <version>3.0</version>
         </dependency>
+
```
Comment (aayushKhandpur) [inline pom.xml]: remove blank line

File: src/main/java/creativei/controller/MainController.java [hunk 2]
```diff
@@ -17,9 +14,24 @@
...
 
     @RequestMapping("/")
     public String index() {
         return "Greetings from Spring Boot!";
     }
+
+    @RequestMapping(value="/pincode/beginWith",produces ="application/json",method = RequestMethod.GET)
+    public @ResponseBody
+    ResponseObject getPincodeStartWith( @RequestParam String pincode){
+      ResponseObject responseObject=  inquiryManager.getPincodes(pincode);
```
Comment (aayushKhandpur) [inline src/main/java/creativei/controller/MainController.java]: Why are we using inquiry manage for pincodes? Use proper abstraction hirarchy

File: src/main/java/creativei/controller/MainController.java [hunk 3]
```diff
@@ -17,9 +14,24 @@
...
+    InquiryManager inquiryManager;
 
     @RequestMapping("/")
     public String index() {
         return "Greetings from Spring Boot!";
     }
+
+    @RequestMapping(value="/pincode/beginWith",produces ="application/json",method = RequestMethod.GET)
+    public @ResponseBody
+    ResponseObject getPincodeStartWith( @RequestParam String pincode){
```
Comment (aayushKhandpur) [inline src/main/java/creativei/controller/MainController.java]: Rename to getMatchingPincodes

File: src/main/java/creativei/controller/MainController.java [hunk 4]
```diff
@@ -17,9 +14,24 @@
...
     BranchManager branchManager;
+    @Autowired
+    InquiryManager inquiryManager;
 
     @RequestMapping("/")
     public String index() {
         return "Greetings from Spring Boot!";
     }
+
+    @RequestMapping(value="/pincode/beginWith",produces ="application/json",method = RequestMethod.GET)
```
Comment (aayushKhandpur) [inline src/main/java/creativei/controller/MainController.java]: rename to /pincodes

File: src/main/java/creativei/controller/MainController.java [hunk 5]
```diff
@@ -17,9 +14,24 @@
...
+    public @ResponseBody
+    ResponseObject getPincodeStartWith( @RequestParam String pincode){
+      ResponseObject responseObject=  inquiryManager.getPincodes(pincode);
+      return responseObject;
+    }
+    @RequestMapping(value = "/pincode/equalsTo",produces = "application/json",method=RequestMethod.GET)
```
Comment (aayushKhandpur) [inline src/main/java/creativei/controller/MainController.java]: We are not getting pincode here, but locality data by pincode. Change mapping to /locality/pincode

File: src/main/java/creativei/controller/MainController.java [hunk 6]
```diff
@@ -17,9 +14,24 @@
...
+    public @ResponseBody
+    ResponseObject getLocalityData(@RequestParam String pincode){
```
Comment (aayushKhandpur) [inline src/main/java/creativei/controller/MainController.java]: Rename to getLocalitiesByPincode

File: src/main/java/creativei/controller/MainController.java (hunk 2 above)
Comment (jitendrajaria) [reply src/main/java/creativei/controller/MainController.java]: Should i make a new Manager Layer for this

File: src/main/java/creativei/dao/CityDao.java [hunk 7]
```diff
@@ -2,9 +2,11 @@
 
 import creativei.entity.City;
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.stereotype.Repository;
 
 import javax.transaction.Transactional;
 
-@Transactional
+@Repository
```
Comment (aayushKhandpur) [inline src/main/java/creativei/dao/CityDao.java]: Should use @Transactional annotation for Dao to make sure Transactions are available

File: src/main/java/creativei/helper/RequestHelper.java [hunk 8]
```diff
@@ -0,0 +1,16 @@
+package creativei.helper;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RequestHelper {
+    private static final Logger logger=LoggerFactory.getLogger(RequestHelper.class);
+    public static Boolean nullStringValidation(String inquiryStr){
```
Comment (aayushKhandpur) [inline src/main/java/creativei/helper/RequestHelper.java]: Should be renamed to isEmptyRequestData(String)

File: src/main/java/creativei/manager/InquiryManager.java [hunk 9]
```diff
@@ -21,4 +18,6 @@
     ResponseObject update(InquiryVo inquiryVo);
     ResponseObject updateAll(List<Inquiry> inquiries);
     ResponseObject getAllEnum();
+    ResponseObject getPincodes(String pincode);
```
Comment (aayushKhandpur) [inline src/main/java/creativei/manager/InquiryManager.java]: Should not be here. Move to LocalityManager

File: src/main/java/creativei/manager/impl/InquiryManagerImpl.java [hunk 10]
```diff
@@ -1,35 +1,34 @@
...
+import java.util.Iterator;
 import java.util.List;
 
 @Service
 public class InquiryManagerImpl implements InquiryManager {
    private static final Logger logger= LoggerFactory.getLogger(InquiryManagerImpl.class);
     @Autowired
     InquiryService inquiryService;
+    @Autowired
+    LocalityService localityService;
```
Comment (aayushKhandpur) [inline src/main/java/creativei/manager/impl/InquiryManagerImpl.java]: Should not be here. Move to LocalityManager

File: src/main/java/creativei/manager/impl/InquiryManagerImpl.java [hunk 11]
```diff
@@ -60,7 +59,9 @@ public ResponseObject create(InquiryVo inquiryVo) {
     }
 
     @Override
-    public ResponseObject getById(Long id) {return null;}
+    public ResponseObject getById(Long id) {
+        return null;
```
Comment (aayushKhandpur) [inline src/main/java/creativei/manager/impl/InquiryManagerImpl.java:54]: Why null return value?

File: src/main/java/creativei/manager/impl/InquiryManagerImpl.java [hunk 12]
```diff
@@ -105,4 +106,36 @@ public ResponseObject getAllEnum() {
         InquiryServerInfoVo inquiryServerInfoVo=new InquiryServerInfoVo();
```
Comment (aayushKhandpur) [inline src/main/java/creativei/manager/impl/InquiryManagerImpl.java:97]: Rename method to getInquiryServerInfo

File: src/main/java/creativei/manager/impl/InquiryManagerImpl.java [hunk 13]
```diff
@@ -105,4 +106,36 @@ public ResponseObject getAllEnum() {
...
     }
+
+    @Override
+    public ResponseObject getPincodes(String pincode) {
+        List<String> pin= localityService.getListOfPincode(pincode);
+        return ResponseObject.getResponse(pin);
+    }
+
+    @Override
+    public ResponseObject getLocalityDataByPincode(String pincode) {
```
Comment (aayushKhandpur) [inline src/main/java/creativei/manager/impl/InquiryManagerImpl.java]: Move to LocalityManager

File: src/main/java/creativei/manager/impl/InquiryManagerImpl.java [hunk 14]
```diff
@@ -105,4 +106,36 @@ public ResponseObject getAllEnum() {
         InquiryServerInfoVo inquiryServerInfoVo=new InquiryServerInfoVo();
         return ResponseObject.getResponse(inquiryServerInfoVo);
     }
+
+    @Override
+    public ResponseObject getPincodes(String pincode) {
```
Comment (aayushKhandpur) [inline src/main/java/creativei/manager/impl/InquiryManagerImpl.java]: Move to LocalityManager

File: src/main/java/creativei/service/impl/LocalityServiceImpl.java [hunk 15]
```diff
@@ -0,0 +1,26 @@
...
+import java.util.List;
+
+@Service("LocalityService")
+public class LocalityServiceImpl implements LocalityService {
+    @Autowired
+    LocalityDao localityDao;
+
+
+    @Override
+    public List<String> getListOfPincode(String pincode) {
```
Comment (aayushKhandpur) [inline src/main/java/creativei/service/impl/LocalityServiceImpl.java]: rename to getPincodes

File: src/main/java/creativei/service/impl/LocalityServiceImpl.java [hunk 16]
```diff
@@ -0,0 +1,26 @@
...
+        return localityDao.findByPincodeStartsWith(pincode);
+    }
+
+    @Override
+    public List<Locality> getLocality(String pincode) {
```
Comment (aayushKhandpur) [inline src/main/java/creativei/service/impl/LocalityServiceImpl.java]: getLocalities

File: src/main/java/creativei/vo/AddressDataVo.java [hunk 17]
```diff
@@ -0,0 +1,44 @@
+package creativei.vo;
+
+import creativei.entity.State;
+
+import java.util.List;
+
+public class AddressDataVo {
```
Comment (aayushKhandpur) [inline src/main/java/creativei/vo/AddressDataVo.java]: Rename to LocalityVo

File: src/main/java/creativei/vo/AddressDataVo.java [hunk 18]
```diff
@@ -0,0 +1,44 @@
...
+    private String country;
+    private StateVo state;
+    private String city;
+    private List<String> locality;
+
```
Comment (aayushKhandpur) [inline src/main/java/creativei/vo/AddressDataVo.java]: add cityId

**Review by aayushKhandpur — CHANGES_REQUESTED** [review]
Address review comments

File: src/main/java/creativei/controller/MainController.java (hunk 2 above)
Comment (aayushKhandpur) [reply src/main/java/creativei/controller/MainController.java]: Yes

File: src/main/java/creativei/vo/AddressDataVo.java (hunk 17 above)
Comment (jitendrajaria) [reply src/main/java/creativei/vo/AddressDataVo.java]: sir there is already a LocalityVo  which has name and id as its fields. AddressDataVo containing the data related to the address.

Comment (aayushKhandpur) [reply src/main/java/creativei/vo/AddressDataVo.java]: Then name it CityVo. We can use it going forward as well.

Comment (jitendrajaria) [reply src/main/java/creativei/vo/AddressDataVo.java]: Sir it contains fields like country and state too CityVo should be related to city only .I have already created the cityVo too which contains city name and its id.

File: pom.xml (hunk 1 above)
Comment (jitendrajaria) [reply pom.xml]: Issues resolves in commit 5f8c87a

File: src/main/java/creativei/controller/MainController.java (hunk 4 above)
Comment (jitendrajaria) [reply src/main/java/creativei/controller/MainController.java]: issues resolved in commit 5f8c87a

File: src/main/java/creativei/controller/MainController.java (hunk 3 above)
Comment (jitendrajaria) [reply src/main/java/creativei/controller/MainController.java]: Issues resolved in commit 5f8c87a

File: src/main/java/creativei/controller/MainController.java (hunk 2 above)
Comment (jitendrajaria) [reply src/main/java/creativei/controller/MainController.java]: Added Locality manager.PR updated in commit 5f8c87a

File: src/main/java/creativei/manager/impl/InquiryManagerImpl.java (hunk 12 above)
Comment (jitendrajaria) [reply src/main/java/creativei/manager/impl/InquiryManagerImpl.java:97]: Issue Resolved

File: src/main/java/creativei/controller/MainController.java (hunk 5 above)
Comment (jitendrajaria) [reply src/main/java/creativei/controller/MainController.java]: Issue Resolved

File: src/main/java/creativei/helper/RequestHelper.java (hunk 8 above)
Comment (jitendrajaria) [reply src/main/java/creativei/helper/RequestHelper.java]: Issue Resolved in commit 5f8c87a

File: src/main/resources/application.properties [hunk 19]
```diff
@@ -31,6 +31,6 @@ spring.jpa.hibernate.naming-strategy = org.hibernate.cfg.ImprovedNamingStrategy
 # Allows Hibernate to generate SQL optimized for a particular DBMS
 spring.jpa.properties.hibernate.dialect = org.hibernate.dialect.MySQL5Dialect
 
-#security.user.name=admin
+#security.user.city=admin
```
Comment (aayushKhandpur) [inline src/main/resources/application.properties:34]: Why are we using these credentials?

**Review by aayushKhandpur — APPROVED** [review]

**Pull request merged**